src/main/java/de/rondiplomatico/nds/floodFill.java

```diff
@@ -0,0 +1,271 @@
+		ff 						= f;
+		minX 					= a;
+		maxX 					= b;
+		minY 					= c;
+		maxY 					= d;
+		dimX 					= maxX - minX + 1;
+		dimY 					= maxY - minY + 1;
+		setStartingPoint();
+	}
+	
+	/*
+	 * Four-way recursive flood fill algorithm
+	 * 
+	 * @param x
+	 * 			the tile x coordinate
+	 * @param y
+	 * 			the tile y coordinate
+	 */
+	public void recursiveFloodFill(int x, int y) {
+		if ((x < 0) || (y < 0) || (x > dimX-1 || (y > dimY-1))) {
+			return;
+		}
+		if (ff[x][y] == bg) {
+			ff[x][y] 			= fill;
+			recursiveFloodFill(x+1, y  ); // right
+			recursiveFloodFill(x,   y+1); // top
+			recursiveFloodFill(x-1, y  ); // left
+			recursiveFloodFill(x,   y-1); // bottom
+		}
+		return;
+	}
+
+	/*
+	 * Eight-way recursive flood fill algorithm
+	 * 
+	 * @param x
+	 * 			the tile x coordinate
+	 * @param y
+	 * 			the tile y coordinate
+	 */
+	public void recursiveFloodFill8(int x, int y) {
+		if ((x < 0) || (y < 0) || (x > dimX-1 || (y > dimY-1))) {
+			return;
+		}
+		if (ff[x][y] == bg) {
+			ff[x][y] 			= fill;
+			recursiveFloodFill(x+1, y  ); // right
+			recursiveFloodFill(x,   y+1); // top
+			recursiveFloodFill(x-1, y  ); // left
+			recursiveFloodFill(x,   y-1); // bottom
+			recursiveFloodFill(x+1, y+1); // top right
+			recursiveFloodFill(x-1, y+1); // top left
+			recursiveFloodFill(x+1, y-1); // bottom right
+			recursiveFloodFill(x-1, y-1); // bottom left
+		}
+		return;
+	}
+
+	/*
+	 * Iterative four-way flood fill algorithm - avoids stack overflow issues
+	 * 
+	 */
+	public void iterativeFloodFill() {
+		Queue<Point> q 			= new LinkedList<Point>();
+		q.add(new Point(startAtX, startAtY));
+		while (!q.isEmpty()) {
+			Point p 			= q.remove();
+			if ((p.x < 0) || (p.y < 0) || (p.x > dimX-1 || (p.y > dimY-1))) {
+				continue;
+			}
+			if (ff[p.x][p.y] == bg) {
+				ff[p.x][p.y] 	= fill;
+				q.add(new Point(p.x+1, p.y  )); // right
+				q.add(new Point(p.x,   p.y+1)); // top
+				q.add(new Point(p.x-1, p.y  )); // left
+				q.add(new Point(p.x,   p.y-1)); // bottom
+			}
+		}
+	}
+
+	/*
+	 * Iterative eight-way flood fill algorithm - avoids stack overflow issues
+	 * 
+	 */
+	public void iterativeFloodFill8() {
+		Queue<Point> q 			= new LinkedList<Point>();
+		q.add(new Point(startAtX, startAtY));
+		while (!q.isEmpty()) {
+			Point p 			= q.remove();
+			if ((p.x < 0) || (p.y < 0) || (p.x > dimX-1 || (p.y > dimY-1))) {
+				continue;
+			}
+			if (ff[p.x][p.y] == bg) {
+				ff[p.x][p.y] 	= fill;
+				q.add(new Point(p.x+1, p.y  )); // right
+				q.add(new Point(p.x,   p.y+1)); // top
+				q.add(new Point(p.x-1, p.y  )); // left
+				q.add(new Point(p.x,   p.y-1)); // bottom
+				q.add(new Point(p.x+1, p.y+1)); // top right
+				q.add(new Point(p.x-1, p.y+1)); // top left
+				q.add(new Point(p.x+1, p.y-1)); // bottom right
+				q.add(new Point(p.x-1, p.y-1)); // bottom left
+			}
+		}
+	}
+	
+	/*
+	 * Safety net for filling left over lines after applying standard flood fill algorithm

```

Comment:
Reading "standard flood fill" -> is there a standard implementation somewhere that's been used? at least give a reference to the implemented algorithm here if so.

Reply:
Nope - I was just referring to the default in this class. I've removed 'standard' to avoid confusion.